src/Cygni.Snake.Client/TileGridBuilder.cs

```diff
@@ -0,0 +1,89 @@
+﻿namespace Cygni.Snake.Client
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    internal class TileGridBuilder
+    {
+        private static readonly Direction[] AllDirections = { Direction.Left, Direction.Right, Direction.Up, Direction.Down };
+        private readonly IDictionary<MapCoordinate, TileType> _tileGrid;
+
+        public TileGridBuilder(int width, int height)
+        {
+            _tileGrid = new Dictionary<MapCoordinate, TileType>();
+
+            for (var x = 0; x < width - 1; x++)
+            for (var y = 0; y < height - 1; y++)
+                _tileGrid[new MapCoordinate(x, y)] = TileType.Empty;
+        }
+
+        public TileGridBuilder WithFood(IEnumerable<MapCoordinate> foodPositions)
+        {
+            foreach (var foodPosition in foodPositions)
+                _tileGrid[foodPosition] = TileType.Food;
+
+            return this;
+        }
+
+        public TileGridBuilder WithObstacles(IEnumerable<MapCoordinate> obstaclePositions)
+        {
+            foreach (var obstaclePosition in obstaclePositions)
+                _tileGrid[obstaclePosition] = TileType.Obstacle;
+
+            return this;
+        }
+
+        public TileGridBuilder WithMySnake(SnakePlayer mySnake)
```

Comment:
This method and WithOpponentSnakes could be refactored to use the same implementation.